process_email.py

```diff
@@ -622,35 +620,50 @@ def _handle_attachment(self, part, file_name, file_path):
                 del attach_meta_info['content']
                 curr_attach['should_ignore'] = True
 
-        part_payload = part.get_payload(decode=True)
-        try:
-            if not part_payload:
-                with open(file_path, 'wb'):
-                    self._base_connector.debug_print("Payload not found, hence adding empty file")
-            else:
+        if isinstance(part.get_payload(), list):
+            try:
+                part_payload = part.get_payload()[0]
+                self._debug_print("payload content type: {0}".format(type(part_payload)))
+                with open(file_path, 'wb') as f:
```

Comment:
Maybe we could write in non-binary mode and skip encoding? This looks like adding extra steps to the process